### PR TITLE
Sample details refactor/ metadata and details fixes

### PR DESCRIPTION
**Pull request by deepsidhu85**
## Description of changes
What did you change in this pull request?  Provide a description of files changed, user interactions changed, etc.  Include how to test your changes.

Updated remaining locations where sample name link pointed to the samples page and replaced with launching the `SampleDetailViewer`. Fixed issue with sample metadata tab which wouldn't set a `projectmetadatarole` for a field if a previous role did not exist. Added logging statements. Updated tests. Refactored UI components to use `Row/Col` from ant design instead of `Space`

## Related issue
Link to the GitHub issue this pull request addresses using the `#issuenum` format.  If it completes an issue, use `Fixes #issuenum` to automatically close the issue.

## Checklist
Things for the developer to confirm they've done before the PR should be accepted:

~* [ ] CHANGELOG.md (and UPGRADING.md if necessary) updated with information for new change.~
~* [ ] Tests added (or description of how to test) for any new features.~
~* [ ] User documentation updated for UI or technical changes.~

### CODE REVIEW

File: src/main/webapp/resources/js/pages/projects/share/ShareAssociated.jsx [hunk 1]
```diff
@@ -40,7 +40,10 @@ export default function ShareAssociated() {
                     </Button>
                   }
                 >
-                  <SampleDetailViewer sampleId={sample.id}>
+                  <SampleDetailViewer
```
Comment (joshsadam) [inline src/main/webapp/resources/js/pages/projects/share/ShareAssociated.jsx:43]: This will have to be updated into the metadata security branch

**Review by joshsadam — CHANGES_REQUESTED** [review]
This is coming along really well.  Thanks for all your hard work.  I am really starting to wonder about this as a modal. There is so much going on here.  Keep going, but we definitely need to talk about it with Eric more.

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UISampleService.java [hunk 2]
```diff
@@ -104,16 +104,22 @@ public UISampleService(SampleService sampleService, ProjectService projectServic
...
-	public SampleDetails getSampleDetails(Long id) {
+	public SampleDetails getSampleDetails(Long id, Long projectId) {
 		Sample sample = sampleService.read(id);
 		Authentication authentication = SecurityContextHolder.getContext()
 				.getAuthentication();
 		boolean isModifiable = updateSamplePermission.isAllowed(authentication, sample);
-
-		return new SampleDetails(sample, isModifiable, cartService.isSampleInCart(id));
+		Project project = null;
+		if (cartService.isSampleInCart(id) != null) {
```
[the remark this answers is not in the document]
Comment (deepsidhu85) [reply src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UISampleService.java:117]: All good! :)

File: src/main/java/ca/corefacility/bioinformatics/irida/repositories/analysis/submission/AnalysisSubmissionRepositoryImpl.java [hunk 3]
```diff
@@ -170,13 +170,13 @@ public AnalysisSubmissionRepositoryImpl(DataSource dataSource,
 			+ "  INNER JOIN analysis_submission_sequencing_object o ON asub.id = o.analysis_submission_id\n"
 			+ "  INNER JOIN sample_sequencingobject sso ON sso.sequencingobject_id = o.sequencing_object_id\n"
 			+ "  INNER JOIN sample s ON sso.sample_id = s.id\n"
-			+ "  LEFT JOIN project_sample psample ON s.id = psample.sample_id\n"
+			+ "  INNER JOIN project_sample psample ON s.id = psample.sample_id\n"
```
[the remark this answers is not in the document]
Comment (deepsidhu85) [reply src/main/java/ca/corefacility/bioinformatics/irida/repositories/analysis/submission/AnalysisSubmissionRepositoryImpl.java:173]: I think so too and probably something that should be updated in dev.

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UISampleService.java [hunk 4]
```diff
@@ -127,7 +133,9 @@ public SampleMetadata getSampleMetadata(Long id, Long projectId) {
 		Sample sample = sampleService.read(id);
 		Set<MetadataEntry> metadataForSample = sampleService.getMetadataForSample(sample);
 
-		List<SampleMetadataFieldEntry> metadata = metadataForSample.stream()
+		List<SampleMetadataFieldEntry> metadata;
+
+		metadata = metadataForSample.stream()
```
[the remark this answers is not in the document]
Comment (deepsidhu85) [reply src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UISampleService.java:138]: I think it was while I was debugging and forgot to revert

File: src/main/java/ca/corefacility/bioinformatics/irida/service/impl/analysis/submission/AnalysisSubmissionServiceImpl.java [hunk 5]
```diff
@@ -349,7 +349,9 @@ public AnalysisSubmission updatePriority(AnalysisSubmission submission, Analysis
 	@Override
 	@PreAuthorize("hasRole('ROLE_ADMIN') or authentication.name == #user.username")
 	public List<ProjectSampleAnalysisOutputInfo> getAllUserAnalysisOutputInfo(User user) {
-		return analysisSubmissionRepository.getAllUserAnalysisOutputInfo(user.getId());
+		final List<ProjectSampleAnalysisOutputInfo> infos = analysisSubmissionRepository.getAllUserAnalysisOutputInfo(user.getId());
+		logger.trace("Found " + infos.size() + " output files for user id=" + user.getId());
```
[the remark this answers is not in the document]
Comment (deepsidhu85) [reply src/main/java/ca/corefacility/bioinformatics/irida/service/impl/analysis/submission/AnalysisSubmissionServiceImpl.java:353]: Yup I just added this in...it was missing the logging statement like the other methods that get the analysis outputs

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/samples/SamplesAjaxController.java [hunk 6]
```diff
@@ -118,7 +119,7 @@ public SamplesAjaxController(UISampleService uiSampleService, UIAnalysesService
 	 * @return {@link SampleMetadata} for the {@link Sample}
 	 */
 	@GetMapping(value = "/{id}/metadata")
-	public ResponseEntity<SampleMetadata> getSampleMetadata(@PathVariable Long id, @RequestParam Long projectId) {
+	public ResponseEntity<SampleMetadata> getSampleMetadata(@PathVariable Long id, Long projectId) {
```
[the remark this answers is not in the document]
Comment (deepsidhu85) [reply src/main/java/ca/corefacility/bioinformatics/irida/ria/web/samples/SamplesAjaxController.java]: Yup totally missed it. Thanks. Updated in cb3fd64ed916db7173ca3b131052aa4a7b73fb90

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UISampleService.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (deepsidhu85) [reply src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UISampleService.java]: Reverted in cb3fd64ed916db7173ca3b131052aa4a7b73fb90

File: src/main/webapp/resources/js/components/activities/ActivityListItem.jsx [hunk 7]
```diff
@@ -54,7 +55,21 @@ export function ActivityListItem({ activity }) {
         // get the item and decide how to add it.
         const item = activity.items[parseInt(fragments[i])];
 
-        if (item.href) {
+        if (
+          parseInt(fragments[i]) === 0 &&
+          (activity.type === "project_sample_data_added" ||
```
[the remark this answers is not in the document]
Comment (deepsidhu85) [reply src/main/webapp/resources/js/components/activities/ActivityListItem.jsx]: Stored in constants in cb3fd64ed916db7173ca3b131052aa4a7b73fb90

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/samples/SamplesAjaxController.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (joshsadam) [reply src/main/java/ca/corefacility/bioinformatics/irida/ria/web/samples/SamplesAjaxController.java]: Interesting that it did not actually cause an issue..

**Pull request merged**